src/za/co/k0ma/playlistWriter/PlaylistWriter.kt

```diff
@@ -36,24 +40,138 @@ class PlaylistWriter(private val files: MutableList<File>, private val outputPat
 					+ "$playlistCounter".format("00") + ".m3u8")
 			
 			outputFile.writeText("$PLAYLIST_HEADER\n")
-			writeEntry(fileList, outputFile)
+			writeEntry(fileList, outputFile, readTags)
 		}
 		
 		println("------ Finished writing files ------");
 	}
 	
-	private fun writeEntry(fileList: List<File>, outputFile: File) {
+	private fun writeEntry(fileList: List<File>, outputFile: File, readTags: Boolean) {
 		fileList.forEach {
-			outputFile.appendText("$ENTRY_START" + createTitle(it) + "\n")
+			outputFile.appendText(ENTRY_START + createTitle(it, readTags) + "\n")
 			outputFile.appendText(it.path + "\n")
 		}
 	}
 	
-	fun createTitle(file: File) = file.nameWithoutExtension
+	private fun createTitle(file: File, readTags: Boolean): String {
```

Comment:
This could be better expressed as:

```
private fun readTag(file: File) = when(file.extension.toLowerCase()) {
        "mp3" -> readID3Tag(file)
    }

private fun createTitle(file: File, readTags: Boolean) =
    if(readTags) readTag(file)
    else file.nameWithoutExtension
```